modules/market/src/main/java/com/opengamma/strata/market/interpolator/LogNaturalDiscountFactorCurveInterpolator.java

```diff
@@ -0,0 +1,232 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.market.interpolator;
+
+import java.io.Serializable;
+
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
+import com.opengamma.strata.math.impl.FunctionUtils;
+import com.opengamma.strata.math.impl.function.PiecewisePolynomialWithSensitivityFunction1D;
+import com.opengamma.strata.math.impl.interpolation.ClampedPiecewisePolynomialInterpolator;
+import com.opengamma.strata.math.impl.interpolation.NaturalSplineInterpolator;
+import com.opengamma.strata.math.impl.interpolation.PiecewisePolynomialResultsWithSensitivity;
+import com.opengamma.strata.math.impl.interpolation.data.Interpolator1DLogPiecewisePoynomialDataBundle;
+import com.opengamma.strata.math.impl.matrix.MatrixAlgebra;
+import com.opengamma.strata.math.impl.matrix.OGMatrixAlgebra;
+
+/**
+ * Find a interpolant F(x) = exp( f(x) ) where f(x) is a Natural cubic spline with Monotonicity cubic filter. 
+ * 
+ * The natural cubic spline is determined by {@link LogNaturalSplineHelper}, where the tridiagonal
+ * algorithm is used to solve a linear system. Since {@link PiecewisePolynomialResultsWithSensitivity}
+ * in {@link Interpolator1DLogPiecewisePoynomialDataBundle} contains information on f(x) (NOT F(x)), 
+ * computation done by {@link PiecewisePolynomialWithSensitivityFunction1D} MUST be exponentiated.
+ */
+final class LogNaturalDiscountFactorCurveInterpolator implements CurveInterpolator, Serializable {
+
+    /**
```

Comment:
Your code settings are indenting by 4 when it should be 2. This should have been fixed by importing the Eclipse code formatter settings, so not sure what has happened.